### PR TITLE
fix(remix): Refactor Cloudflare env checks for clarity

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Description

Tiny refactor to simplify the types/checks

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Review by panteliselef — APPROVED** [review]
LGTM

**Comment by nikosdouvlis** [comment]
> @nikosdouvlis I think that we should add a changeset that indicates that this is breaking as we move the `Cloudflare pages` check before the Node one.

I was about to change it back exactly because of itbeing an breaking change, so good job spotting that :)

File: packages/remix/src/utils/utils.ts [hunk 1]
```diff
@@ -24,8 +24,15 @@ export function assertValidClerkState(val: any): asserts val is ClerkState | und
...
-  env: Record<string, string>;
+type CloudflareEnv = { env: Record<string, string> };
+
+// https://remix.run/blog/remix-vite-stable#cloudflare-pages-support
+const hasCloudflareProxyContext = (context: any): context is { cloudflare: CloudflareEnv } => {
+  return !!context?.cloudflare?.env;
+};
+
+const hasCloudflareContext = (context: any): context is CloudflareEnv => {
+  return !!context?.env;
```
Comment (dimkl) [inline packages/remix/src/utils/utils.ts:35]: ❓ I think this check is not CloudflareContext specific. Isn't the `context.env` populated in other cases too?

**Review by dimkl — APPROVED** [review]

File: packages/remix/src/utils/utils.ts (hunk 1 above)
Comment (nikosdouvlis) [reply packages/remix/src/utils/utils.ts:35]: Afaik, only CF prepopulates .env

**Comment by nikosdouvlis** [comment]
@octoper please take another look

**Comment by octoper** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @octoper - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 1.0.1-snapshot.v4542e9d |
| @clerk/chrome-extension | 1.0.1-snapshot.v4542e9d |
| @clerk/clerk-js | 5.0.1-snapshot.v4542e9d |
| @clerk/clerk-expo | 1.0.1-snapshot.v4542e9d |
| @clerk/fastify | 1.0.1-snapshot.v4542e9d |
| gatsby-plugin-clerk | 5.0.1-snapshot.v4542e9d |
| @clerk/localizations | 2.0.1-snapshot.v4542e9d |
| @clerk/nextjs | 5.0.1-snapshot.v4542e9d |
| @clerk/clerk-react | 5.0.1-snapshot.v4542e9d |
| @clerk/remix | 4.0.1-snapshot.v4542e9d |
| @clerk/clerk-sdk-node | 5.0.1-snapshot.v4542e9d |
| @clerk/shared | 2.0.1-snapshot.v4542e9d |
| @clerk/themes | 2.0.1-snapshot.v4542e9d |
| @clerk/types | 4.0.1-snapshot.v4542e9d |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/backend`
```sh
npm i @clerk/backend@1.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.1-snapshot.v4542e9d --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.0.1-snapshot.v4542e9d --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.0.1-snapshot.v4542e9d --save-exact
```

**Review by octoper — APPROVED** [review]

**Pull request merged**